src/BrandwatchReactAuth.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import jwtDecode from 'jwt-decode';
+import TokenStore from 'viziaauth';
+
+export default class BrandwatchReactAuth extends Component {
+  getChildContext() {
+    return {
+      brandwatchAuthLogout: this.handleLogout,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = new TokenStore(this.props.domain),
+    this.handleLogout = this.handleLogout.bind(this);
+
+    this.state = {
+      loggedIn: undefined,
+    };
+
+    props.onCreateStore(this.store);
+  }
+
+  componentWillMount() {
+    this.store.getToken({ aud: this.props.audience }).then((token) => {
+      if (!token) {
+        return window.location.href = this.store.loginUrl;
+      }
+
+      this.props.onAuthenticate({ token, profile: jwtDecode(token) });
+      this.setState(() => ({ loggedIn: true }));
+    });
+  }
+
+  handleLogout() {
+    return this.store.removeToken({ aud: this.props.audience }).then(() => {
+      location.href = this.store.loginUrl;
```

Comment:
Is `window.location` the same as `location`, we are using both in here (I wonder if this component needs to consider server side rendering?)